2017/proposals/ranasinghe-kanchana.md

```diff
@@ -0,0 +1,156 @@
+* norm basic functionality including tests and documentation
+
+With regards to the normalization function, in a Matplotlib sense, it encompasses a function mapping different data values to different colors (or color levels). Currently, the different data values to be mapped from are expected to be numerical. This limitation will be worked around to enable support for categorical data (string data). Also during the course of the project focus will be given to the currently available discrete normalization methods (matplotlib.colors.BoundaryNorm) which handle data in a different sense which may be used as a foundation for this additional unit support. 
+
+### July 3rd - July 7th
+* framework outline for cmap functionality extensions
+* cmap basic functionality including tests and documentation
+
+Currently, the process of using matplotlib.pyplot for heat-maps and other scalarMappables includes the creation of a color-map followed by normalization of all available data to fit this color-map. Color-maps plotting functionality is threefold: sequential, divergent and qualitative. Considering the intrinsic discreteness of categorical data, the extension of the color-maps API will focus on its qualitative plotting ability. However the fundamental behavior of the color-maps API will not require any changes for handling categorical data as our approach simply involves the integration of unit support. Therefore it is possible to build this extended support for categorical data while keeping the public API intact. 
+
+### July 10th - July 14th
+* integration of norm and cmap into heatmaps
+
+Given the functionality extended for categorical data, the final task is entwining all this functionality to imshow to allow the plotting of heat-maps from non-numerical values. This would be handled here. 
+
+### July 17th - July 21th, **End of Phase 2**
+* testing and debugging of norm and cmap APIs
+
+### July 24th - July 28th, **Begin of Phase 3**
+* identifying possible scalarMappables to extend categorical color support
+* framework outline for categorical color support of other scalarMappables
+
+Identification of other scalarMappables like matshow, pcolor, pcolormesh, scatter, etc. will be done here. Possibilities to extend the current work to support categorical data will be analyzed. The work here will not be direct extensions of heat-maps but will require different approaches to extend the units support for these scalarMappables.  
+
+### July 31st - August 4th
+* basic functionality of other scalarMappables 
+
+### August 7th - August 11th
+* overall functionality of other scalarMappables with tests and documentation
+
+### August 14th - August 18th
+* testing and debugging of other scalarMappables code
+
+### August 21st - August 25th, **Final Week**
+* testing of all code and bug fixing
+* final review of all code and documentation
+
+### August 28th - August 29th, **Submit final work**
+* code submission
```

Comment:
Code should be coming in through incremental PRs all summer.  Doing a massive injest at the end will make it very hard to review and greatly increase the chances that there will be conflicts on the merge (both of which will drastically reduce the chance that you code ends up merged and shipped).